packages/ember-data/lib/system/record_arrays/many_array.js

```diff
@@ -74,10 +74,11 @@ export default Ember.Object.extend(Ember.MutableArray, Ember.Evented, {
       return record.get('isNew');
     });
     toSet = toSet.concat(newRecords);
-    this.arrayContentWillChange(0, this.length, this.length);
+    var oldLength = this.length;
+    this.arrayContentWillChange(0, oldLength, toSet.length);
     this.set('length', toSet.length);
     this.currentState = toSet;
-    this.arrayContentDidChange(0, this.length, this.length);
+    this.arrayContentDidChange(0, oldLength, toSet.length);
```

Comment:
could'nt it be this.arrayContentWillChange(0, oldLength, this.length); ?
I think this is ok because now, this.length refers to the 'new' length.